### PR TITLE
Include LCSC part number when grouping parts.

### PR DESCRIPTION
**Pull request by ademuri**
I'm working on a design which uses multiple LED colors, with the same
footprint. Without this change, I get the error message:

[ERROR] Components D4, D3, D2 from same group have different LCSC part numbers:                 C2286, C72043

### CODE REVIEW

File: jlc_kicad_tools/jlc_lib/generate_bom.py [hunk 1]
```diff
@@ -55,14 +55,7 @@ def GenerateBOM(input_filename, output_filename, opts):
...
-            # Get the field name for the LCSC part number.
-            for field_name in c.getFieldNames():
-                field_value = c.getField(field_name).strip()
-
-                if LCSC_PART_NUMBER_MATCHER.match(field_value):
-                    lcsc_part_number = field_value
+            lcsc_part_number = c.getLcscPartNumber()
 
             if lcsc_part_number:
                 lcsc_part_numbers.add(lcsc_part_number)
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/generate_bom.py:61]: Can you simplify the logic below now that non-uniqueness is no longer possible? It seems like lcsc_part_numbers (the list) no longer needs to exist

Comment (matthewlai) [reply jlc_kicad_tools/jlc_lib/generate_bom.py:61]: Also the assume_same_lcsc_partnumber would no longer work, so I would just remove it

**Review by matthewlai — CHANGES_REQUESTED** [review]
Thanks!

File: jlc_kicad_tools/jlc_lib/generate_bom.py (hunk 1 above)
Comment (ademuri) [reply jlc_kicad_tools/jlc_lib/generate_bom.py]: Simplified this and removed that flag. I also changed the behavior of `--warn-no-lcsc-partnumber` to still warn when `--include-all-component-groups` is set - does the change I made make sense?

Comment (matthewlai) [reply jlc_kicad_tools/jlc_lib/generate_bom.py]: Perfect thanks!

File: jlc_kicad_tools/jlc_lib/generate_bom.py [hunk 2]
```diff
@@ -48,62 +48,27 @@ def GenerateBOM(input_filename, output_filename, opts):
...
-                lcsc_part_numbers_none_found = True
+            # All components in a group should have the same part number
+            lcsc_part_number = c.getLcscPartNumber()
 
             if c.getFootprint() != "":
                 footprints.add(c.getFootprint())
 
-        # Check part numbers for uniqueness
-        if len(lcsc_part_numbers) == 0 and not opts.include_all_groups:
+        if lcsc_part_number == None:
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/generate_bom.py:63]: Can you use "lcsc_part_number is None" for this?

Comment (ademuri) [reply jlc_kicad_tools/jlc_lib/generate_bom.py]: Yes, done.

**Comment by matthewlai** [comment]
Amazing thanks!

**Pull request merged**